messaging/repository/room_repository.go

```diff
@@ -0,0 +1,210 @@
+	DeleteRoom(ctx context.Context, roomID string) error
+}*/
+
+type RoomRepository struct {
+	dbSession *gocql.Session
+}
+
+func NewRoomRepository(session *gocql.Session) *RoomRepository {
+	return &RoomRepository{
+		dbSession: session,
+	}
+}
+const (
+	SaveRoom = `INSERT INTO chat.room (roomID, NAME, Admin, students) VALUES (?, ?, ?,?);`
+	GetRoom = `SELECT * FROM chat.room WHERE roomID=? LIMIT 1;`
+	GetRoomFor = `SELECT * FROM chat.student_rooms WHERE student=? LIMIT 1 ;`
+	editChatroomParticipant = `UPDATE chat.room  SET Students=? WHERE RoomID=?;`
+
+	addRoomForParticipant = `UPDATE chat.student_rooms  SET rooms = rooms +? WHERE student=? ;`
+	CreateStudentRooms=`INSERT INTO chat.student_rooms (student) VALUES (?);`
+	RemoveRoomForParticipant = `UPDATE chat.student_rooms  SET rooms = rooms-? WHERE student=? ;`
+
+)
+
+
+func (r RoomRepository) SaveRoom( room *domain.ChatRoom) error {
+	var students []string
+	for _, element := range room.Students{
+		students = append(students, element.ID)
+	}
+
+	 err := r.dbSession.Query(SaveRoom, room.RoomID, room.Admin.ID,room.Name,students).Consistency(gocql.One).Exec();
+
+	if err!=nil {
+		log.Fatal(err)
+	}
+
+
+	fmt.Println("Message entry: ", room.RoomID, room.Admin,room.Name,room.Students)
+
+	return err
+
+}
+
+
+func (r RoomRepository) GetRoom( roomID string) (*domain.ChatRoom, error) {
+	var room domain.ChatRoom
+
+	var studentText  []string
+	var AllStudent []domain.Student
+
+	err := r.dbSession.Query(GetRoom, roomID).Consistency(gocql.One).Scan(&room.RoomID, &room.Admin.ID,&room.Deleted, &room.Name, &studentText);
+
+
+	if err!=nil {
+		log.Fatal(err)
+		return nil,err ;
+	}
+
+	for _,ID := range studentText{
+		var student domain.Student
+		student.ID=ID
+		AllStudent = append(AllStudent,student )
+
+	}
+	room.Students= AllStudent
+
+
+
+
+
+	return &room,err
+
+}
+
+
+func (r RoomRepository) EditChatRoomParticipants( roomID string, student []domain.Student) error {
+var studentID []string
+
+	for _,s := range student{
+
+		studentID = append(studentID,s.ID )
+
+	}
+	err := r.dbSession.Query(editChatroomParticipant, studentID, roomID).Consistency(gocql.One).Exec();
+	if err!=nil {
+		log.Fatal(err)
+		return err ;
+	}
+
+	return  nil
+}
+
+func (r RoomRepository) GetRoomsFor( studentID string) (*domain.StudentChatRooms, error) {
+	var StudentRoom domain.StudentChatRooms
+	var roomsID []string
+	var rooms []domain.ChatRoom
+	err := r.dbSession.Query(GetRoomFor, studentID).Consistency(gocql.One).Scan(&StudentRoom.Student.ID, &roomsID);
+	if err!=nil {
+		log.Fatal(err)
+		return nil,err ;
+	}
+	for _,ID := range roomsID{
+		var room domain.ChatRoom
+		room.RoomID=ID
+		rooms = append(rooms,room )
+
+	}
+	StudentRoom.Rooms=rooms
+
+	return &StudentRoom , err
+}
+
+
+
+func (r RoomRepository) AddRoomForParticipants( roomID string, students []domain.Student) error {
+	var rooms []string
+	rooms=append(rooms,roomID)
+
+	for _,student:= range students {
+
```

Comment:
we should check that each student exists before adding the room for a student, if at least 1 doesn't exist, then return a message: cannot complete transaction bc not all students exist and exit normally, no panic or fatal exit

Reply:
done in the usecase layer and removed the fatal exit ( it was just easier to debug then )